qcodes/data/hdf5_format.py

```diff
@@ -307,7 +321,7 @@ def _read_list_group(self, entry_point, list_type):
         elif list_type == 'list':
             item = [d[k] for k in sorted(d.keys())]
         else:
-            raise Exception('type %s not supported' % type(item))
+            raise Exception('type %s not supported' % type(list_type))
```

Comment:
as far as i read the code `list_type` is actually a string, so perhaps the type call is not needed
```suggestion
            raise Exception('type %s not supported' % list_type)
```